parsers/IE.py

```diff
@@ -0,0 +1,258 @@
+    """
+    assert type(target_datetime) == datetime
+    assert kind != ""
+
+    r = session or Session()
+    resp: Response = r.get(
+        KINDS_URL[kind].format(
+            dt=target_datetime.strftime("%d-%b-%Y"), zone=ZONE_MAPPING[zone_key]["key"]
+        )
+    )
+    try:
+        data = resp.json().get("Rows", {})
+    except:
+        raise ParserException(
+            parser="IE.py",
+            message=f"{target_datetime}: {kind} data is not available for {zone_key}",
+        )
+    filtered_data = [item for item in data if item["Value"] is not None]
+    return filtered_data
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_production(
+    zone_key: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """Gets values for wind production and estimates unknwon production as demand - wind - exchange"""
+    if target_datetime is None:
+        target_datetime = arrow.utcnow().datetime
+
+    demand_data = fetch_data(
+        target_datetime=target_datetime,
+        zone_key=zone_key,
+        kind="demand",
+        session=session,
+    )
+    wind_data = fetch_data(
+        target_datetime=target_datetime, zone_key=zone_key, kind="wind", session=session
+    )
+    exchange_data = fetch_data(
+        target_datetime=target_datetime,
+        zone_key=zone_key,
+        kind="exchange",
+        session=session,
+    )
+    assert len(demand_data) > 0
+    assert len(wind_data) > 0
+    assert len(exchange_data) > 0
+
+    production = []
+    for item in demand_data:
+        dt = item["EffectiveTime"]
+        wind_dt = [item for item in wind_data if item["EffectiveTime"] == dt]
+        if len(wind_dt) == 1:
+            wind_prod = wind_dt[0]["Value"]
+        else:
+            wind_prod = 0
+        exchange_dt = [item for item in exchange_data if item["EffectiveTime"] == dt]
+        if len(exchange_dt) == 1:
+            exchange = exchange_dt[0]["Value"]
+        else:
+            exchange = 0
+        data_point = {
+            "zoneKey": "IE",
+            "datetime": datetime.strptime(dt, "%d-%b-%Y %H:%M:%S").replace(
+                tzinfo=pytz.timezone("Europe/Dublin")
+            ),
+            "production": {
+                "unknown": item["Value"] - exchange - wind_prod,
+                "wind": wind_prod,
+            },
+            "source": "eirgridgroup.com",
+        }
+        production += [data_point]
+    return production
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_exchange(
+    zone_key1: str,
+    zone_key2: str,
+    session: Optional[Session] = None,
+    target_datetime: Optional[datetime] = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    """gets exchanges values for the East-West interconnector (GB->IE)"""
+    if target_datetime is None:
+        target_datetime = arrow.utcnow().datetime
```

Comment:
I prefer using the datetime lib whenever possible but I guess that's a personal prefrence.